src/tallies/tally_scoring.cpp

```diff
@@ -2468,4 +2465,63 @@ void score_surface_tally(Particle& p, const vector<int>& tallies)
     match.bins_present_ = false;
 }
 
+void score_pulse_height_tally(Particle& p, const vector<int>& tallies)
+{
+
+  for (auto i_tally : tallies) {
+    auto& tally {*model::tallies[i_tally]};
+
+    // Initialize an iterator over valid filter bin combinations.  If there are
+    // no valid combinations, use a continue statement to ensure we skip the
+    // assume_separate break below.
+    auto filter_iter = FilterBinIter(tally, p);
+
+    auto i_energy_filt = tally.filters()[tally.energy_filter_];
+    auto i_energy_bin = p.filter_matches(i_energy_filt).i_bin_;
+    auto bin_energy = p.filter_matches(i_energy_filt).bins_[i_energy_bin];
+
+    const EnergyFilter& energy_filt {
+      *dynamic_cast<EnergyFilter*>(model::tally_filters[i_energy_filt].get())};
+
+    auto i_cell_filt = tally.filters()[tally.cell_filter_];
+    auto i_cell_bin = p.filter_matches(i_cell_filt).i_bin_;
+    auto bin_cell = p.filter_matches(i_cell_filt).bins_[i_cell_bin];
+
+    const CellFilter& cell_filt {
+      *dynamic_cast<CellFilter*>(model::tally_filters[i_cell_filt].get())};
+
+    const auto& cells = cell_filt.cells();
+    int cell_counter = 0;
+    for (auto cell_id : cells) {
+      p.filter_matches(i_cell_filt).bins_[i_cell_bin] = cell_counter;
+      double score = p.pht_storage()[cell_id];
+
+      if (score < energy_filt.bins().front() ||
+          score > energy_filt.bins().back()) {
+        continue;
+      } else {
+        auto i_match = lower_bound_index(
+          energy_filt.bins().begin(), energy_filt.bins().end(), score);
+        p.filter_matches(i_energy_filt).bins_[i_energy_bin] = i_match;
```

Comment:
Is it not possible to just utilize the existing EnergyFilter logic here? This is effectively replicating what `EnergyFilter` already does.

Reply:
Discussed during monthly drop in session May

Reply:
Ok, so I know we talked through this and I said it's not possible, but after thinking about it a bit more, we might actually be able to hijack the normal logic of `CellFilter` and `EnergyFilter` by temporarily changing the cell/energy of the particle. Something like:
```C++
// Save original cell/energy information
int orig_n_coord = p.n_coord();
int orig_cell = p.coord(0).cell;
int orig_E_last = p.E_last();

for (auto cell_id : cells) {
  // Temporarily change cell
  p.n_coord() = 1;
  p.coord(0).cell = cell_id;

  // Temporarily change energy
  ...
  p.E_last() = p.pht_storage()[index];

  // Get scoring filter combinations
  auto filter_iter = FilterBinIter(tally, p);

  // Normal logic for accumulating tally
  ...
}
// Restore cell/energy
p.n_coord() = orig_n_coord;
p.coord(0).cell = orig_cell;
p.E_last() = orig_E_last;
```

Reply:
As said, this is a very cool idea